lib/chef/knife/ec2_server_create.rb

```diff
@@ -852,9 +875,10 @@ def validate!
           exit 1
         end
 
-        if config[:security_group_ids] && config[:security_group_ids].class == String
+        # Validation for security_group_ids passed through knife.rb. It will raise error if values are not provided in Array.
+        if locate_config_value(:security_group_ids) && locate_config_value(:security_group_ids).class == String
           ui.error("Invalid value type for knife[:security_group_ids] in knife configuration file (i.e knife.rb). Type should be array. e.g - knife[:security_group_ids] = ['sgroup1']")
-          exit 1
+          exit
```

Comment:
I have updated the code. The specs where failing when I was using exit 1. Found the issue and fixed the specs.